manifests/0000_26_cloud-controller-manager-operator_16_credentialsrequest-gcp.yaml

```diff
@@ -15,10 +15,107 @@ spec:
   providerSpec:
     apiVersion: cloudcredential.openshift.io/v1
     kind: GCPProviderSpec
-    predefinedRoles:
-      - roles/compute.instanceAdmin
-      - roles/compute.loadBalancerAdmin
-      - roles/iam.serviceAccountUser
+    permissions:
+    - "compute.networks.get"
+    - "compute.networks.list"
+    - "compute.networks.use"
+    - "compute.zones.list"
+    - "resourcemanager.projects.get"
+    - "compute.subnetworks.list"
+    - "compute.subnetworks.get"
+    - "compute.subnetworks.use"
+    - "compute.backendServices.create"
+    - "compute.backendServices.delete"
+    - "compute.backendServices.get"
+    - "compute.backendServices.list"
+    - "compute.backendServices.update"
+    - "compute.backendServices.use"
+    - "compute.disks.create"
+    - "compute.disks.createSnapshot"
+    - "compute.disks.delete"
+    - "compute.disks.get"
+    - "compute.disks.list"
+    - "compute.disks.listEffectiveTags"
+    - "compute.disks.listTagBindings"
+    - "compute.disks.resize"
+    - "compute.disks.setLabels"
+    - "compute.disks.update"
+    - "compute.disks.use"
+    - "compute.disks.useReadOnly"
+    - "compute.forwardingRules.create"
+    - "compute.forwardingRules.delete"
+    - "compute.forwardingRules.get"
+    - "compute.forwardingRules.list"
+    - "compute.forwardingRules.update"
+    - "compute.forwardingRules.use"
+    - "compute.instanceGroups.create"
+    - "compute.instanceGroups.delete"
+    - "compute.instanceGroups.get"
+    - "compute.instanceGroups.list"
+    - "compute.instanceGroups.update"
+    - "compute.instanceGroups.use"
+    - "compute.regionBackendServices.create"
+    - "compute.regionBackendServices.delete"
+    - "compute.regionBackendServices.get"
+    - "compute.regionBackendServices.getIamPolicy"
+    - "compute.regionBackendServices.list"
+    - "compute.regionBackendServices.setIamPolicy"
+    - "compute.regionBackendServices.setSecurityPolicy"
+    - "compute.regionBackendServices.update"
+    - "compute.regionBackendServices.use"
+    - "compute.instances.attachDisk"
+    - "compute.instances.create"
```

Comment:
I'm not sure that CCM needs `instance.create` permission...

Reply:
I think it might need it according to this code: https://github.com/openshift/cloud-provider-gcp/blob/master/providers/gce/gce_instances.go#L492-L498

Reply:
This thing uses within tests. I don't think that we need to request this permission for production installations.

I suggest to double check any create/delete permissions. Some of the functions which you pointed might not be used at all or be used in a tests code.